cmd/csi/main.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright © 2018-2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/kubernetes-csi/csi-lib-iscsi/iscsi"
+	"github.com/openebs/jiva-csi/pkg/config"
+	"github.com/openebs/jiva-csi/pkg/driver"
+	"github.com/openebs/jiva-csi/pkg/kubernetes/client"
+	"github.com/openebs/jiva-csi/version"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/klog"
+	k8scfg "sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+// log2LogrusWriter implement io.Writer interface used to enable
+// debug logs for iscsi lib
+type log2LogrusWriter struct {
+	entry *logrus.Entry
+}
+
+// Write redirects the std log to logrus
+func (w *log2LogrusWriter) Write(b []byte) (int, error) {
+	n := len(b)
+	if n > 0 && b[n-1] == '\n' {
+		b = b[:n-1]
+	}
+	w.entry.Debug(string(b))
+	return n, nil
+}
+
+var enableISCSIDebug bool
+
+/*
+ * main routine to start the jiva-csi-driver. The same
+ * binary is used for controller and agent deployment.
+ * they both are differentiated via plugin command line
+ * argument. To start the controller, we have to pass
+ * --plugin=controller and to start it as node, we have
+ * to pass --plugin=node.
+ */
+func main() {
+	var config = config.Default()
+	// initializing klog for the kubernetes libraries used
+	klog.InitFlags(nil)
+	cmd := &cobra.Command{
+		Use:   "jiva-csi-driver",
+		Short: "driver for provisioning jiva volume",
+		Long:  `provisions and deprovisions the volume`,
+		Run: func(cmd *cobra.Command, args []string) {
+			run(config)
+		},
+	}
+
+	cmd.Flags().AddGoFlagSet(flag.CommandLine)
+	_ = flag.CommandLine.Parse([]string{})
+
+	cmd.PersistentFlags().StringVar(
+		&config.NodeID, "nodeid", "", "NodeID to identify the node running this driver",
```

Comment:
why this is empty? we should fill the node name here.

Reply:
Node name will be filled from YAML.